app/src/pages/counterPage/containers/CounterContainer.js

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react'
+import CounterView from '../components/CounterView'
+
+class CounterContainer extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            countValue:0,
+            isEven: true,
+        };
+    }
+
+    handleIncrement = () => {
+        this.setState( (state) => {
+            const incrementedValue = state.countValue + 1;
+            const isEven = incrementedValue %2 ===0;
+
+            return {
+                countValue: incrementedValue,
+                isEven,
+            };
+        });
+    }
+
+    handleDecrement = () => {
+        if (this.state.countValue>0) {
+            this.setState( (state) => {
+                const decrementedValue = state.countValue - 1;
+                const isEven = decrementedValue %2 ===0;
```

Comment:
Put your logic of parity calculation to componentDidUpdate to avoid duplicate code fragments